library/src/scala/Selectable.scala

```diff
@@ -34,3 +34,18 @@ object Selectable:
   implicit def reflectiveSelectableFromLangReflectiveCalls(x: Any)(
       using scala.languageFeature.reflectiveCalls): scala.reflect.Selectable =
     scala.reflect.Selectable.reflectiveSelectable(x)
+
+  /** A marker trait for subclasses of `Selectable` indicating
+   *  that precise parameter types are not needed for method dispatch. That is,
+   *  a class inheriting from this trait and implementing
+   *
+   *     def applyDynamic(name: String, paramTypes: Class[_]*)(args: Any*)
+   *
+   *  should dispatch to a method with the given `name` without having to rely
+   *  on the precise `paramTypes`. Subtypes of `SignatureCanBeImprecise`
+   *  can have more relaxed subtyping rules for refinements. They do not need
+   *  the additional restriction that the signatures of the refinement and
+   *  the definition that implements the refinment must match.
+   */
+  trait WithoutPreciseParameterTypes extends Selectable
```

Comment:
This needs to be `@experimental` or commented out until 3.1

Reply:
I moved that part to a separate PR #12268 with a 3.1 milestone.